tendrl/commons/objects/cluster/atoms/check_cluster_nodes_up/__init__.py

```diff
@@ -0,0 +1,76 @@
+import etcd
+import json
+
+from tendrl.commons.flows.exceptions import FlowExecutionFailedError
+from tendrl.commons.objects import BaseAtom
+from tendrl.commons.utils import etcd_utils
+from tendrl.commons.utils import log_utils as logger
+
+
+class CheckClusterNodesUp(BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(CheckClusterNodesUp, self).__init__(*args, **kwargs)
+
+    def run(self):
+        try:
+            all_node_status_up = True
+            # check job is parent or child
+            job = NS.tendrl.objects.Job(
+                job_id=self.parameters['job_id']
+            ).load()
+            if "parent" not in job.payload:
+                # fetch node id using integration_id
+                integration_id = self.parameters[
+                    'TendrlContext.integration_id'
+                ]
+                key = "indexes/tags/tendrl/integration/%s" % \
+                    integration_id
+                node_ids_str = etcd_utils.read(key).value
+                node_ids = json.loads(node_ids_str)
+                # identifying node status using node_id
+                hosts = {}
+                for node in node_ids:
```

Comment:
combine the 2 for loops, why do you need 2 loops?

Reply:
When import flow is started then first for loop will collect list of node_id of particular cluster, so using list of node_id i can display a message what are all the nodes i am going to check, 
like 
Checking status of nodes[A, B]

In second for loop i am checking from list of node which node is up and which node is down

so first for loop is only to collect list of nodes to display a single message

other wise it leads to multiple job updates

Reply:
This explanation is not useful at all.  Combine the 2 loops and display that single job update based on 
`node_ids = json.loads(node_ids_str)`

I am not sure why you cant do these things via 1 loop, this isnt something we should spend time debating about

Reply:
i will change this :)

Reply:
i found what is the mistake i did here, no need two for loop